### PR TITLE
Add version compatibility information

### PR DESCRIPTION
**Pull request by mosabua**
## Description

- for the CLI and the JDBC driver

## Additional context and related issues

Discussed on slack at https://trinodb.slack.com/archives/C01TEP0HJTH/p1673554134517209

## Release notes

(x) This is not user-visible or docs only and no release notes are required.
( ) Release notes are required, please propose a release note for me.
( ) Release notes are required, with the following suggested text:

### CODE REVIEW

**Review by willshen — APPROVED** [review]
Thanks for adding this.

An added bonus (not necessary) would be to document what's regularly tested. From what I can see, we test 5 versions for every 3 versions back (not sure how to better word this...) https://github.com/trinodb/trino/blob/master/testing/trino-test-jdbc-compatibility-old-server/src/test/java/io/trino/TestJdbcResultSetCompatibilityOldServer.java

File: docs/src/main/sphinx/client/cli.rst [hunk 1]
```diff
@@ -16,6 +16,11 @@ It can be used with Java version 8 and higher.
 The CLI uses the :doc:`Trino client REST API </develop/client-protocol>` over
 HTTP/HTTPS to communicate with the coordinator on the cluster.
 
+The CLI version should be identical to the version of the Trino cluster, or
+newer. Older versions typically work, but only a subset is regularly tested.
```
Comment (colebow) [inline docs/src/main/sphinx/client/cli.rst:20]: I feel like this first sentence is awkward. "The CLI version should be up-to-date with or newer than the Trino cluster version" might be a little smoother? Or something along those lines.

File: docs/src/main/sphinx/client/jdbc.rst [hunk 2]
```diff
@@ -16,6 +16,11 @@ The Trino JDBC driver has the following requirements:
 * All users that connect to Trino with the JDBC driver must be granted access to
   query tables in the ``system.jdbc`` schema.
 
+The JDBC driver version should be identical to the version of the Trino cluster,
```
Comment (colebow) [inline docs/src/main/sphinx/client/jdbc.rst:19]: Same here.

**Review by colebow — APPROVED** [review]
LGTM, feel free to ignore the comment about wording, it's reasonable as-is.

File: docs/src/main/sphinx/client/cli.rst (hunk 1 above)
Comment (mosabua) [reply docs/src/main/sphinx/client/cli.rst:20]: ... I inserted this and reread ... "up-date-with or newer" feels worse than what there is in the PR now.

Comment (colebow) [reply docs/src/main/sphinx/client/cli.rst:20]: Maybe just replace "identical" with "up-to-date"?

**Comment by mosabua** [comment]
Maybe for merge after the release @martint

File: docs/src/main/sphinx/client/cli.rst [hunk 3]
```diff
@@ -16,6 +16,11 @@ It can be used with Java version 8 and higher.
...
+Specifically ensure that you do not mix versions from before and after Trino
```
[the remark this answers is not in the document]
Comment (mosabua) [reply docs/src/main/sphinx/client/cli.rst:21]: Done

File: docs/src/main/sphinx/client/jdbc.rst [hunk 4]
```diff
@@ -16,6 +16,10 @@ The Trino JDBC driver has the following requirements:
...
+or newer. Older versions typically work, but only a subset is regularly tested.
+Versions before 350 are not supported.
```
Comment (mosabua) [inline docs/src/main/sphinx/client/jdbc.rst:21]: Also done here fyi @electrum

**Comment by mosabua** [comment]
Ready again @electrum

**Review by ebyhr — APPROVED** [review]

**Review by electrum — APPROVED** [review]

**Pull request merged**